tests/chainer_tests/functions_tests/array_tests/test_rollaxis.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(
+    {'axis': 0, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': 2, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': 1, 'start': 1, 'out_shape': (2, 3, 4)},
+    {'axis': -3, 'start': 2, 'out_shape': (3, 2, 4)},
+    {'axis': -1, 'start': 0, 'out_shape': (4, 2, 3)},
+    {'axis': -2, 'start': -2, 'out_shape': (2, 3, 4)},
```

Comment:
I agree with you in that we do not have to test trivial cases nor to test those cases which are essentially same. But I believe the test cases I listed runs the different code path, except one combination mention below.

Namely, we have combination of the following branchings 
* `start` is positive or negative
* `axis` is positive or negative
* (modified) `start` is bigger or smaller than (modified) `axis`.
Each test case reflects one of each situation listed.

 `(axis, start) = (0, 2)` and `(axis, start) = (1, 1)` runs the same code path. So we can safely remove these cases.